frontend/src/metabase/common/components/EntityPicker/components/NestedItemPicker/AutoScrollBox.tsx

```diff
@@ -0,0 +1,73 @@
+import type React from "react";
+import { useEffect, useRef } from "react";
+import { usePrevious } from "react-use";
+
+import { HorizontalScrollBox } from "./NestedItemPicker.styled";
+
+const gradualScroll = (container: HTMLDivElement | null) => {
+  if (!container) {
+    return;
+  }
+  const intervalId = setInterval(() => {
+    // This is actually a fairly fussy check. It's possible for the scrollWidth to be higher
+    // than what the scrollLeft + clientWidth can ever be, causing you to constantly scroll
+    // right.
+    if (
+      container.scrollLeft + container.clientWidth <=
+      container.scrollWidth * 0.99
+    ) {
+      container.scrollLeft += 25;
+    } else {
+      clearInterval(intervalId);
+    }
+  }, 10);
```

Comment:
Would `element.scrollIntoView({ behavior: 'smooth', inline: 'start' });` be useful here, or is it not the right effect?

https://codesandbox.io/p/sandbox/horizontal-scrolling-qlrty9

Reply:
🤔 I'll give it a try

Reply:
Probably better than the derpy "JS scroll" I put together here haha.

Reply:
@iethree @npfitz What do you think? Though Nick's code is rock solid, I was thinking it might improve DX to use the browser's native powers.

Reply:
I like the scroll into view idea better than what we have here. I'll try to give it a go later today

Reply:
my misgiving is the grossness of having to assign a ref to the last child and pass it around. maybe it's not that gross 🤷

Reply:
The reason I like it is because on more than 1 occasion, the entity picker has decided to always scroll left for me (infinite loop of scrolling left). Happened mostly when I'm on battery and things are slowing down on my machine. Using scrollIntoView would likely remove that possibility from ever happening